src/orders/types/desired-order-item.type.ts

```diff
@@ -0,0 +1,22 @@
+import { ApiProperty } from '@nestjs/swagger';
+
+export class DesiredOrderItem {
```

Comment:
**suggestion(non-blocking):** this felt like a way to avoid mapping between the domain and dto types (and in order to use it in the service layer). I would have avoided it personally using the mappers